### DISCUSSION TITLE
Proper configuration for manually creating a reactivated project?

### DISCUSSION THREAD

**Discussion by mmonj** [silviogutierrez/reactivated#274] [Q&A] [answered]
Hello there, 

I've followed the [instructions](https://www.reactivated.io/documentation/existing-projects/) for extending an existing project with reactivated. So far I've been trying to set up a clean project while following the manual instructions before I move on to applying the same setup steps onto my desired existing project.

This is the repo I currently have set up: https://github.com/mmonj/reactivated-skeleton, but the template `TestPage.tsx` file that I created keeps getting the error: `TS2305: Module '"@reactivated"' has no exported member 'templates'.`

Is there anything that you can see that I may be missing that is the reason that the prop types aren't being generated?

Thanks in advance

![IMG-9116](https://github.com/silviogutierrez/reactivated/assets/68614941/afd5669a-1e97-4d44-8424-82f2df71e282)

**Comment by mmonj** [comment, marked as answer]
It looks like I made the grave mistake of not including the app into the main `urls.py` file. Big derp. Sorry for the bother.

**Reply by silviogutierrez** [reply]
No problem! If you need more help just let me know.

**Comment by Adamcina** [comment]
@mmonj  As I can see from the screenshot, you are on Windows? Did you use Nix installer or Docker to install Reactivated? 
Could you please help me with installing Reactivated because I'm also on Windows but I'm having a hard time making Reactivated work, even with Docker (never used nix or docker before).
I've tried to configure the existing project, but this is an error I get:
```
 File "C:\Users\User\AppData\Local\Programs\Python\Python311\Lib\site-packages\reactivated\apps.py", line 182, in generate_schema
    process = subprocess.Popen(
              ^^^^^^^^^^^^^^^^^
  File "C:\Users\User\AppData\Local\Programs\Python\Python311\Lib\subprocess.py", line 1024, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\User\AppData\Local\Programs\Python\Python311\Lib\subprocess.py", line 1493, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 2] The system cannot find the file specified
```
Also, I tried with another PC and fresh installation of python 3.9.0 and installing specific versions of Django (4.0.0) and Node (16.16) in virtualenvironment, but it didn't help either, I get the same error as above.

Following steps with Docker, the command `docker run -itp 8000:8000 -v ${PWD}:/app --name my_app silviogutierrez/reactivated nix-shell` gives me this error:
`error: getting status of '/app/default.nix': No such file or directory`

**Comment by mmonj** [comment]
@Adamcina I am on windows. I decided to go with my own manual installation method based on Reactivated's official setup [instructions](https://www.reactivated.io/documentation/existing-projects/).

You can check out my [boilerplate repo](https://github.com/mmonj/reactivated-skeleton) and just run my `docker-compose.yml` to get the container started. Make sure to edit the user/UID to match that of your host system.

Run `docker compose up`, enter the dev container in vscode and run `python manage.py runserver 0.0.0.0:8000` in the container's shell to start the server. That should be all there is to it

**Reply by Adamcina** [reply]
@mmonj Thanks for the help, I appreciate it! As I've never worked with Docker before, I didn't manage to solve some problems with Docker that came up, so I ended up using Virtual Machine and Ubuntu.

**Reply by mmonj** [reply]
@Adamcina So you were able to get the container running? What issues with docker did you have specifically? The best way to use docker on windows is to install WSL2 first, then run the docker desktop installer and tell it to use wsl2

**Reply by Adamcina** [reply]
@mmonj  

> So you were able to get the container running?

Yes, but only for the first time. After that, I wasn't able to run the container anymore and was getting this error:
`error: getting status of '/app/default.nix': No such file or directory`
This probably doesn't have anything to do with Docker but with my lack of experience with Docker. So, I didn't want to waste time anymore on investigating and solving the issue, it was a a lot easier for me to start Linux VM and get it working. 
Anyway, I plan to get into Nix, as Silvio suggested, because I see that is incredibly useful so it certainly deserves more attention.
